### PR TITLE
feat: User can keyword filter within artist auction results

### PR DESCRIPTION
**Pull request by olerichter00**
Addresses [CX-1591]

Allows users to filter artist auction results by keyword.

|Screenshot 2021-07-09 at 13 13 50|Screenshot 2021-07-09 at 13 15 52|
|---|---|

|Screenshot 2021-07-09 at 13 15 26|
|---|

[CX-1591]: https://artsyproduct.atlassian.net/browse/CX-1591

### CODE REVIEW

File: src/v2/Apps/Artist/Routes/AuctionResults/Components/SortSelect.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const SortSelect = () => {
         <Select {...props} />
       </Media>
       <Media greaterThan="xs">
-        <Select {...props} title="Sort" />
+        <Select {...props} title="Sort:" />
```
Comment (olerichter00) [inline src/v2/Apps/Artist/Routes/AuctionResults/Components/SortSelect.tsx:40]: **Note:** I adjusted the title to the newest design.

File: src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx [hunk 2]
```diff
@@ -192,7 +204,7 @@ const AuctionResultsContainer: React.FC<AuctionResultsProps> = ({
             toggleMobileActionSheet={toggleMobileActionSheet}
           />
 
-          <Spacer mt={3} />
+          <Spacer mt={["2", "0"]} />
```
Comment (olerichter00) [inline src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx:207]: **Note:** We need more space on mobile.

File: src/v2/Apps/Artist/Routes/AuctionResults/Components/KeywordFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { LabeledInput, MagnifyingGlassIcon } from "@artsy/palette"
+import { debounce } from "lodash"
+import React, { useEffect, useMemo, useRef } from "react"
+import { useAuctionResultsFilterContext } from "../AuctionResultsFilterContext"
+
+const DEBOUNCE_DELAY = 300
```
Comment (olerichter00) [inline src/v2/Apps/Artist/Routes/AuctionResults/Components/KeywordFilter.tsx:6]: **Note:** The debounce delay can be adjusted, but 300 seems to make sense to me.

File: src/v2/Apps/Artist/Routes/AuctionResults/Components/AuctionResultsControls.tsx [hunk 4]
```diff
@@ -23,9 +24,12 @@ export const AuctionResultsControls = ({ artist, toggleMobileActionSheet }) => {
                 Filter
               </Flex>
             </Button>
-            <SortSelect />
+            <Flex ml={1}>
+              <SortSelect />
+            </Flex>
```
Comment (olerichter00) [inline src/v2/Apps/Artist/Routes/AuctionResults/Components/AuctionResultsControls.tsx:29]: **Note:** Added some margin here to create some space between the select and the filter button on small screens.

File: src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx (hunk 2 above)
Comment (damassi) [reply src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx:207]: Shouldn't need to wrap these values in strings per rest of file.

File: src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx [hunk 5]
```diff
@@ -172,12 +173,23 @@ const AuctionResultsContainer: React.FC<AuctionResultsProps> = ({
...
       )}
       <Media greaterThan="xs">
-        <Flex justifyContent="space-between" alignItems="flex-start" pb={4}>
-          <Text variant="xs" textTransform="uppercase">
-            Filter by
-          </Text>
-          <SortSelect />
-        </Flex>
+        <Row>
+          <Col sm={3} pr={[0, 2]}>
```
Comment (damassi) [inline src/v2/Apps/Artist/Routes/AuctionResults/ArtistAuctionResults.tsx:177]: These column components are deprecated in favor of [GridColumns](https://github.com/artsy/palette/blob/master/packages/palette/src/elements/GridColumns/GridColumns.story.tsx).

(Are you using VSCode? If so, it should cross out these components as deprecated,)

**Review by damassi — APPROVED** [review]
One request around swapping out deprecated components, but otherwise looks really good 👍

**Comment by pvinis** [comment]
For me it looks good, so I'll leave it to Chris' notes.

**Comment by olerichter00** [comment]
#squashongreen

**Pull request merged**